mqtt/mqtt.js

```diff
@@ -80,14 +80,27 @@ function startMqtt() {
     return net.createServer(aedes.handle).listen(config.mqtt_port);
 }
 
-nats.subscribe('channel.*', {'queue':'mqtts'}, function (msg) {
+function parseMqttTopic(topic) {
+    // allow channels/any-char-allowed/message with 0|+ block /any-char-allowed
+    // the text messages and the last subtopic can not terminate with /
+    return /^channels\/(.+?)\/messages(\/.+[^\/]$)*$/.exec(topic)
+}
+
+nats.subscribe('channel.>', {'queue':'mqtts'}, function (msg) {
     var m = message.RawMessage.decode(Buffer.from(msg)),
-        packet;
+        packet, channelParts;
     if (m && m.protocol !== 'mqtt') {
+        // rexexp has two element, the first one is the channelId, wile
+        // the second is undefined or the subtopic parts starting from the dot
+        channelParts = /^([\w\-]+)(\..+[^\.]$)*$/.exec(m.channel)
         packet = {
             cmd: 'publish',
             qos: 2,
-            topic: 'channels/' + m.channel + '/messages',
+            // compose the topic with channelId followed by /messages and
+            // all the subtopics in the form /subtopic/subN/
+            topic: 'channels/' + channelParts[1] + '/messages' + (
```

Comment:
What if `channelParts` has length 1?

Reply:
Good catch there!

Reply:
@anovakovic01 afsik if the regexp match it will be always of length 3, first element contains the full string, second the channeid, and the third could be undefined or the subtopic part

Reply:
@beres Sorry, I missed this check later.